pyomo/core/base/set.py

```diff
@@ -1457,17 +1383,145 @@ def clear(self):
 
     def set_value(self, val):
         self.clear()
-        for x in val:
-            self.add(x)
+        self.update(val)
+
+    def _initialize(self, val):
+        self.update(val)
 
     def update(self, values):
-        for v in values:
-            if v not in self:
-                self.add(v)
+        # _values was initialized above...
+        #
+        # Special case: set operations that are not first attached
+        # to the model must be constructed.
+        if isinstance(values, SetOperator):
+            values.construct()
+        try:
+            val_iter = iter(values)
+        except TypeError:
+            logger.error(
+                "Initializer for Set %s%s returned non-iterable object "
+                "of type %s."
+                % (
+                    self.name,
+                    ("[%s]" % (index,) if self.is_indexed() else ""),
+                    (values if values.__class__ is type else type(values).__name__),
+                )
+            )
+            raise
+
+        if self._dimen is not None:
+            if normalize_index.flatten:
+                val_iter = self._cb_normalized_dimen_verifier(self._dimen, val_iter)
+            else:
+                val_iter = self._cb_raw_dimen_verifier(self._dimen, val_iter)
+        elif normalize_index.flatten:
+            val_iter = map(normalize_index, val_iter)
+        else:
+            val_iter = self._cb_check_set_end(val_iter)
+
+        if self._domain is not Any:
+            val_iter = self._cb_domain_verifier(self._domain, val_iter)
+
+        if self._filter is not None:
+            val_iter = filter(partial(self._filter, self.parent_block()), val_iter)
+
+        if self._validate is not None:
+            val_iter = self._cb_validate(self._validate, self.parent_block(), val_iter)
+
+        # We wrap this check in a try-except because some values
+        #  (like lists) are not hashable and can raise exceptions.
+        try:
+            self._update_impl(val_iter)
+        except Set._SetEndException:
+            pass
 
     def pop(self):
         return self._values.pop()
 
+    def _cb_domain_verifier(self, domain, val_iter):
```

Comment:
What does the `cb` in these verifier/validator method names mean?

Reply:
"callback"